jailer.go

```diff
@@ -426,6 +426,24 @@ func LinkFilesHandler(kernelImageFileName string) Handler {
 				m.Cfg.Drives[i].PathOnHost = String(driveFileName)
 			}
 
+			if m.Cfg.hasSnapshot() {
```

Comment:
`hasSnapshot` returns true if either [snapshotPath or memFilePath is non-empty,](https://github.com/firecracker-microvm/firecracker-go-sdk/blob/d7a9b5ea4a5150bfdeda6ab9a20812f2657aac79/machine.go#L174) so we should check if the individual file paths are non-empty, before creating the file link.